R/validate_stan_model.R

```diff
@@ -18,10 +19,12 @@ validate_stan_model <- function(
   pkpdsim_model,
   data,
   mapping = NULL,
+  add_parameters = NULL,
```

Comment:
I believe we removed this in a previous PR, but we do still need it IMO. Reason is that "parameters" don't really resemble the same thing in PKPDposterior vs PKPDsim. In PKPDposterior a parameters is a value that is always sampled, it cannot be fixed or switched off. In PKPDsim a parameter can be either fixed or estimated. So we do need the ability to add parameters when switching to PKPDsim.

Reply:
I think this is a pretty big philosophical decision. Maybe we can talk about it at our next meeting?

The problem I see with doing it the way you have it here is that it is difficult to track why a model might not pass validate_stan_model. For example, in your code, you've passed `TH_REGI = -0.2` as an argument to `add_parameters`. On the other hand, in the `parameter_definitions` the expression for CL includes a hardcoded -0.2 factor. It's possible that these two wouldn't align via human error.

As an alternative, can't we add TH_REGI as a parameter and simply not estimate it? See for example how we implemented `TH_CRCL` in the example [pk_vanco_thomson model.](https://github.com/InsightRX/PKPDposterior/blob/main/inst/models/pk_vanco_thomson.stan) 

This would reduce complexity for validate_stan_model and produce models that are generally better aligned with each other. On the other hand, it would potentially make parse_model_definitions/new_stand_model a little more complicated since you'd need to designate a set of parameters as fixed. However, this is true for how we handle it in PKPDsim too.

Reply:
Yes, I think that approach would be a bit nicer. Let me explore if that works and what caveats are.